activerecord/test/cases/scoping/named_scoping_test.rb

```diff
@@ -453,17 +469,17 @@ def test_table_names_for_chaining_scopes_with_and_without_table_name_included
 
   def test_scopes_with_reserved_names
     class << Topic
-      def public_method; end
-      public :public_method
+      def attribute_was; end
+      public :attribute_was
 
-      def protected_method; end
-      protected :protected_method
+      def attribute_changed?; end
+      protected :attribute_changed?
 
-      def private_method; end
-      private :private_method
+      def changed; end
+      private :changed
     end
 
-    [:public_method, :protected_method, :private_method].each do |reserved_method|
+    [:attribute_was, :attribute_changed?, :changed].each do |reserved_method|
```

Comment:
If you use `method_defined_within?` as I suggested above, these won't need to change.